timur/lib/client/jsx/components/browser/browser.jsx

```diff
@@ -67,21 +77,62 @@ export default function Browser({model_name, record_name, tab_name}) {
   );
   const {view, record, tab, revision, template, can_edit} = browserState;
   const [mode, setMode] = useState('loading');
-  const loading = !view || !template || !record || !tab_name;
+  const [error, setError] = useState(null);
   const {cancelEdits, approveEdits} = useEditActions(setMode, browserState);
-  const {selectOrShowTab, selectDefaultTab, selectTab, showTab} = useTabActions(
-    browserState,
-    setMode
-  );
+  const requestDocuments = useRequestDocuments();
+
+  const browseToTab = useCallback((tabName) => {
+    invoke(setLocation(window.location.href.replace(/#.*/, '') + `#${tabName}`));
+  }, [invoke, setLocation]);
 
   // Set at 'skin' on the browser styling.
   let skin = 'browser';
   if (mode === 'browse') skin = 'browser ' + model_name;
   const editMode = useCallback(() => setMode('edit'), [setMode]);
 
-  // On mount
+  // const loading = !view || !template || !record || !tab_name;
+  const [loading, loadDocuments] = useAsyncWork(function* loadDocuments(model_name, view) {
```

Comment:
`useAsyncWork` was a hook I introduced for the search page, too.  It basically uses es6 generator expressions to "flatten" promise chains.  It is similar to expanding each yielded promise with a `.then(() =>` except for two main benefits:
1.  It supports cancelling.  Promises do not (natively)
2. It's a flatter syntax that more elegantly handles native js syntax (like loops, ifs, try, etc).